source/magpie/group.d

```diff
@@ -17,7 +17,13 @@ struct Group(GrpRowType...)
     /// number of elements before the particular group
     int[] elementCountTill;
 
-    alias GrpType = staticMap!(toArr, GrpRowType);
+    /// Flag to check if the Group is Homogeneous
+    enum bool isHomogeneousType = isHomogeneous!(GrpRowType);
+
+    static if(!isHomogeneous!(GrpRowType))
+        alias GrpType = staticMap!(toArr, GrpRowType);
+    else
+        alias GrpType = toArr!(GrpRowType[0])[GrpRowType.length];
```

Comment:
this code is duplicate with dataframe, move it to utils, and do 
```
alias GrpType = homogeneousTypesFor!(GrpRowType);
```

if you need `isHomogeneousType` use a mixin template.